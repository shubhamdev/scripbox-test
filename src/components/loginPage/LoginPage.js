import React, { useContext, useEffect, useRef } from "react";
import { Form, Input, Button } from "antd";
import { Row, Col } from "antd";
import { login, isLogin } from "../../utils";
import { PageContext } from "../../PageContextProvider";
const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 20 },
};
const tailLayout = {
  wrapperCol: { offset: 8, span: 16 },
};

const validateMessages = {
  required: "${label} is required!",
};

function LoginPage(props) {
  const inputRef = useRef(null);
  const { user } = useContext(PageContext);
  const onFinish = (value) => {
    login(value);
    user[1](value);
    props.history.push("/dashboard");
  };

  useEffect(() => {
    if (isLogin()) {
      props.history.push("/dashboard");
    } else {
      inputRef.current.focus();
    }
  }, []);

  return (
    <Row style={{ padding: "10%" }} justify="center">
      <h1 id="test">Login Form</h1>
      <Col span={12}>
        <Form
          {...layout}
          name="nest-messages"
          onFinish={onFinish}
          validateMessages={validateMessages}
        >
          <Form.Item
            name={["user", "name"]}
            label="User Name"
            rules={[{ required: true }]}
          >
            <Input
              ref={inputRef}
              type="text"
              name="username"
              autoComplete="off"
              placeholder="Enter e-mail and phone number"
            />
          </Form.Item>
          <Form.Item {...tailLayout}>
            <Button id="submit-id" type="primary" htmlType="submit">
              Login
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </Row>
  );
}

export default LoginPage;
