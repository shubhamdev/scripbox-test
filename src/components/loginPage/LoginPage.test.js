import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";
import LoginPage from "./LoginPage";

let realUseContext;
let useContextMock;

// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
  jest.resetAllMocks();
  // cleanup();
});

// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

describe("LoginPage component", () => {
  it("renders without crashing", () => {
    useContextMock.mockReturnValue("User");
    const element = new ShallowRenderer().render(<LoginPage />);
    expect(element.props.children).toMatchSnapshot();
  });
});
