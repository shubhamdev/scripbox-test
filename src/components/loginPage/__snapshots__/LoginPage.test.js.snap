// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders without crashing 1`] = `
Array [
  <h1
    id="test"
>
    Login Form
</h1>,
  <Unknown
    span={12}
>
    <Unknown
        labelCol={
            Object {
                "span": 8,
              }
        }
        name="nest-messages"
        onFinish={[Function]}
        validateMessages={
            Object {
                "required": "\${label} is required!",
              }
        }
        wrapperCol={
            Object {
                "span": 20,
              }
        }
    >
        <FormItem
            label="User Name"
            name={
                Array [
                    "user",
                    "name",
                  ]
            }
            rules={
                Array [
                    Object {
                      "required": true,
                    },
                  ]
            }
        >
            <Input
                autoComplete="off"
                name="username"
                placeholder="Enter e-mail and phone number"
                type="text"
            />
        </FormItem>
        <FormItem
            wrapperCol={
                Object {
                    "offset": 8,
                    "span": 16,
                  }
            }
        >
            <Unknown
                htmlType="submit"
                id="submit-id"
                type="primary"
            >
                Login
            </Unknown>
        </FormItem>
    </Unknown>
</Unknown>,
]
`;
